tests/unit/src/main/java/org/mule/tck/junit4/rule/RequiresConnectivity.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tck.junit4.rule;
+
+import static org.junit.Assume.assumeTrue;
+
+import org.junit.rules.ExternalResource;
+
+import java.io.IOException;
+import java.net.URL;
+
+/**
+ * Checks that a connection can be established to a provided Url. If not, the test case is ignored.
+ *
+ * @since 4.1
+ */
+public class RequiresConnectivity extends ExternalResource {
```

Comment:
Why you use a rule instead of an assumption?

Reply:
just for the code to be cleaner if this is to be reused.